Library/Homebrew/cask/dsl.rb

```diff
@@ -486,7 +486,7 @@ def depends_on(**kwargs)
     def add_implicit_macos_dependency
       return if (cask_depends_on = @depends_on).present? && cask_depends_on.macos.present?
 
-      depends_on macos: ">= :#{MacOSVersion::SYMBOLS.key MacOSVersion::SYMBOLS.values.min}"
+      depends_on macos: ">= #{MacOSVersion::SYMBOLS.keys.last.inspect}"
```

Comment:
Very cautious about fragility of this. If we want to assume order like this, it should be in a helper function with unit tests.

Reply:
```suggestion
      depends_on macos: ">= :#{MacOSVersion::SYMBOLS.key MacOSVersion::SYMBOLS.values.min}"
```
lets revert this back for now

Reply:
I believe this code is intended to target the oldest allowed macOS version, so what do we think about `">= #{MacOSVersion.new(HOMEBREW_MACOS_OLDEST_ALLOWED).to_sym.inspect}`? We use `HOMEBREW_MACOS_NEWEST_UNSUPPORTED` as a `MacOSVersion.new` argument in places, so there's precedent for this general idea.